axelor-base/src/main/java/com/axelor/apps/base/service/pricing/PricingServiceImpl.java

```diff
@@ -98,16 +123,45 @@ public List<Pricing> getPricings(
       bindings.put("modelName", modelName);
     }
 
-    if (previousPricing != null) {
-      filter.append("AND self.previousPricing = :previousPricing ");
-      bindings.put("previousPricing", previousPricing);
+    if (pricing != null) {
+      filter.append("AND self.linkedPricing = :linkedPricing ");
+      bindings.put("linkedPricing", pricing);
     } else {
-      filter.append("AND self.previousPricing is NULL ");
+      filter.append("AND self.linkedPricing is NULL ");
     }
+
+    filter.append("AND (self.archived = false OR self.archived is null) ");
+
+    appendProductFilter(product, productCategory, filter, bindings);
+    LOG.debug("Filtering pricing with {}", filter);
+    return pricingRepo.all().filter(filter.toString()).bind(bindings).fetch();
+  }
+
+  @Override
+  public List<Pricing> getAllPricings(
+      Company company, Product product, ProductCategory productCategory, String modelName) {
+    LOG.debug("Fetching pricings");
+    StringBuilder filter = new StringBuilder();
+    Map<String, Object> bindings = new HashMap<>();
+
+    filter.append("self.startDate <= :todayDate ");
+    filter.append("AND (self.endDate > :todayDate OR self.endDate = NULL) ");
+    bindings.put("todayDate", appBaseService.getTodayDate(company));
+
+    if (company != null) {
+      filter.append("AND self.company = :company ");
+      bindings.put("company", company);
+    }
+
+    if (modelName != null) {
+      filter.append("AND self.concernedModel.name = :modelName ");
+      bindings.put("modelName", modelName);
+    }
```

Comment:
I don't like the code duplication, I see 2 ways to manage this:
* factorize in a separate protected methods the filter and bindings
* replace directly getRootPricingForNextPricings with a unique query to directly fetch the root pricing